src/libs/ajax.js

```diff
@@ -1226,6 +1226,27 @@ const Surveys = signal => ({
   })
 })
 
+const WorkspaceDataService = signal => ({
```

Comment:
Can we split this out into a separate file following the pattern of ajax/Apps.js, ajax/AzureStorage.js, etc?

Reply:
Separating it would also allow writing it in TypeScript and having types for these functions would remove some of the need to declare types where these functions are called in WDSDataTableProvider.

Reply:
yes, definitely have this be broken out to /libs/ajax/WorkspaceDataService.js or better yet /libs/ajax/WorkspaceDataService.ts.  I also second Nick's recommendation to 'promote' some of the data related types elsewhere to this .ts file.

nit:  Should the name be WorkspaceData to align with the other namings?

Reply:
also, once broken out, the methods need unit tests (with mocked fetch to avoid real network calls).  Having the broken out file be .ts will help with type safety on the unit tests :)